pkg/build/inline.go

```diff
@@ -122,12 +122,68 @@ var nonDNS = regexp.MustCompile(`[^-a-z0-9\.]`)
 // ComponentFiles reads file-references for component builder objects.
 // The returned components are copies with the file-references removed.
 func (n *Inliner) ComponentFiles(ctx context.Context, comp *bundle.ComponentBuilder) (*bundle.Component, error) {
-	name := comp.ComponentName
+	newObjs, err := n.objectFiles(ctx, comp.ObjectFiles, comp.ComponentReference())
+	if err != nil {
+		return nil, err
+	}
+
+	newObjs, err = n.objectTemplateBuilders(ctx, newObjs, comp.ComponentReference())
```

Comment:
Does this add a file indirection to ComponentBuilder. I.e. ComponentBuilder file -> ObjectTemplateBuilder file -> template? Since ObjectTemplateBuilders are fairly concise, it would be nice to represent them inlined in ComponentBuilder.

Reply:
you mean like the following?

```go
type ComponentBuilder struct {
  ObjectFiles []File
  ObjectTemplates []TemplateBuilders
}

type TemplateBuilders {
  ObjectMeta
  TemplateType string
  File File
}
```

That's possible. I sort of like the modularity of the ObjectTemplateBuilders -- we largely just rely on the same tech, with the caveat there's some hooks in the inlining code. Maybe that caveat is significant enough to rethink that.